### DISCUSSION TITLE
Create a datasource or dashboard variable type that can set variable values based on simple logic operations

### DISCUSSION THREAD

**Discussion by ethirolle** [grafana/grafana#38197] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
A new dashboard variable type (or a new "datasource" type) that allows one to calculate variables values based on simple logic operations.

**Why is this needed**:
We have a customer who is using a clever approach to calculate values for some hidden variables based on the value chosen for a single visible variable picker.

I.e., the user picks the value of the one visible variable, then a Query-type variable uses a SQL datasource to execute this simple select...case logic to determine the value of a second variable:

```
select
  case
    when '$longwindow' = '1h' then '5m'
    when '$longwindow' = '6h' then '30m'
    when '$longwindow' = '1d' then '2h'
    when '$longwindow' = '3d' then '6h'
  else 'invalid'
end
```

This feature request is to ask for some way to accomplish this same goal without the use of a SQL datasource to do if/then or select/case logic operations.

**Other Notes:**
This sounds somewhat similar to the interval variable and it’s auto mode, but is not exactly the same. The approach described above (with select/case logic) allows for more fine-tuned control over exact intervals chosen.

**Comment by hugohaggmark** [comment]
@ethirolle not sure this is something we'll prioritise any time soon.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
